### DISCUSSION TITLE
OCAPI get geolocation

### DISCUSSION THREAD

**Discussion by chuan0** [SalesforceCommerceCloud/sfcc-ci#452] [Team Posts]
Is there a way to get the request (I need geolocation specifically) in OCAPI hooks?

**Comment by aventurinisalesforce** [comment]
it should be not hide on the hook

**Comment by chuan0** [comment]
@aventurinisalesforce so how do you access the request variable then? in storefront controllers, it's passed into the function, the hook only passes the referenced object(dw.ocapi.shop.order.afterPOST passes in order, for example)

**Comment by saurabh-jain012** [comment]
request object will be available directly like request.locale or request.clientId.

**Comment by chuan0** [comment]
Makes sense, thanks!

**Comment by hnestmann** [comment]
> Is there a way to get the request (I need geolocation specifically) in OCAPI hooks?

https://github.com/SalesforceCommerceCloud/sfcc-hooks-collection/pull/25/files
